### ISSUE TITLE
Issues with array JSON content

### ISSUE THREAD

**Issue by dmoeyersons** [Alfresco/alfresco-js-api#861]
**Type of issue:**
```
- [ ] New feature request
- [x] Bug  
- [ ] Support request
```
**Current behavior:**

- I open a json file containing a json array using the ADF Service ContentService, function getNodeContent.
Content of the json file: [PASCI-Productieprocessen-basisprocessen.txt](https://github.com/Alfresco/alfresco-js-api/files/5200316/PASCI-Productieprocessen-basisprocessen.txt)

- In the subscribe action I get the following error message and the file contents is not returned:
![image](https://user-images.githubusercontent.com/41050239/92699225-9bca9e00-f34d-11ea-9210-8c313cfdc3e7.png)

**Expected behavior:**
- I open a json file containing a json array using the ADF Service ContentService, function getNodeContent.
Content of the json file: [PASCI-Productieprocessen-basisprocessen.txt](https://github.com/Alfresco/alfresco-js-api/files/5200316/PASCI-Productieprocessen-basisprocessen.txt)
- In the subscribe action the json content is returned. (Worked in 3.3.0)

**Steps to reproduce the issue:**
- Rename the attached file PASCI-Productieprocessen-basisprocessen.txt to PASCI-Productieprocessen-basisprocessen.json
- Upload it to Alfresco
- Read the content in ADF 4.0.0 using the ADF Service ContentService, function getNodeContent

**Node version (for build issues):**  
v.12.16.2

**Alfresco JS API version**
4.0.0

**Extra information:**
The error happens in the deserialize function in alfrescoApiClient.ts.
The data is extracted correctly from the data response (data = response.text). The data variable now contains an array.
As you can see, the returnType is 'function Blob()'
- The first check returnType==='blob' fails ('function Blob()' is not 'blob')
- The second check Array.isArray(data) is positive.
each element is converted to a new Blob. The elements on its own contain complex object data. The Blob conversion fails because the elements on its own aren't arrays.

![image](https://user-images.githubusercontent.com/41050239/92700903-a71ec900-f34f-11ea-8298-5419d0c4937d.png)

**Comment by eromano** [comment]
can you chek if this is https://github.com/Alfresco/alfresco-js-api/pull/1080 related?

**Comment by eromano** [comment]
Solved in ADF 4.4.0
